java/org/contikios/cooja/Simulation.java

```diff
@@ -150,68 +155,133 @@ public Simulation(Cooja cooja, long seed) {
     this.cooja = cooja;
     randomGenerator = new SafeRandom(this);
     randomSeed = seed;
-  }
-
-  @Override
-  public void run() {
-    assert isRunning : "Did not set isRunning before starting";
-    lastStartRealTime = System.currentTimeMillis();
-    lastStartSimulationTime = getSimulationTimeMillis();
-    speedLimitLastRealtime = lastStartRealTime;
-    speedLimitLastSimtime = lastStartSimulationTime;
+    this.simulationThread = new Thread(() -> {
+      boolean isSimulationRunning = false;
+      boolean isAlive = true;
```

Comment:
Seems to only be used inside the do-loop, push it to the same level as nextEvent is declared.

Reply:
Good point. I will move down `isSimulationRunning`.